.github/workflows/container-tests.yml

```diff
@@ -3,13 +3,12 @@ on:
     types:
       - created
 
-concurrency:
-  group: container_tests-${{ github.event.issue.number }}
-  cancel-in-progress: true
-
 jobs:
   container-tests:
     runs-on: ubuntu-20.04
+    concurrency:
+      group: container-${{ github.event.issue.number }}-${{ matrix.version }}-${{ matrix.os_test }}
+      cancel-in-progress: true
     name: "Container-tests: ${{ matrix.version }} - ${{ matrix.os_test }}"
```

Comment:
You could also move `name` to the beginning of the job, so it is consistent across the workflows.